### PR TITLE
Bl-6139 smmt api key encryption

### PR DESCRIPTION
**Pull request by Marcin-Duszynski**
### What are the relevant Jira tickets?
Bl-6139

### Where should the reviewer start?
src/config/smmt

## Dev checklist:
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Has code been tested manually?
- Has code been covered by unit tests?
- Are unit test executed on Jenkins during pull request build?
- Do all unit tests pass?
- Is Jira ticket state updated?

## Code review checklist (for reviewers -> Copy to your comment)
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Do you understand task goal and business context?
- Is documentation updated and understandable?
- Has code been covered by unit tests?
- Did you identify technical issues/debt and reported it to TA? 
- Does code meet SOLID principles?
- Does code meet DRY principle?
- Is code in better state than before (scout rule)?

### CODE REVIEW

File: app/src/smmt/client.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ function getSmmtResponseSuperagent(marque, vin) {
 
 function vincheck(marque, vin) {
   logger = loggerFactory.create();
+  logger.debug(config, 'SMMT api configuration');
```
Comment (ryszardm-kainos) [inline app/src/smmt/client.js]: I'd refrain from printing the config with the api key, even on debug level

File: app/test/config/smmt.test.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+        promise: () => Promise.resolve({ Plaintext: 'apiKey' }),
+      }),
+    };
+  },
+};
 
 describe('When SMMT config loader is called', () => {
-  it('then SMMT uri and api key is returned', () => {
-    const config = configLoader.load();
+  it('then SMMT uri and api key is returned', (done) => {
```
Comment (ryszardm-kainos) [inline app/test/config/smmt.test.js]: maybe one test for when kms decrypt fails? ;)

File: app/src/smmt/client.js (hunk 1 above)
Comment (Marcin-Duszynski) [reply app/src/smmt/client.js]: sorry, it should be removed before PR

File: app/test/config/smmt.test.js (hunk 2 above)
Comment (Marcin-Duszynski) [reply app/test/config/smmt.test.js]: done

**Review by ryszardm-kainos — DISMISSED** [review]

File: app/src/config/smmtConfigurationLoader.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+      return data.Plaintext.toString('ascii');
+    })
+    .catch((reason) => {
+      logger.error({ context: reason }, 'SMMT api key decryption failed.');
+      return undefined;
+    });
+}
+
+exports.load = () => co.wrap(function* loadConfig(key) {
```
Comment (przemos) [inline app/src/config/smmtConfigurationLoader.js]: loadConfig(key) does not seem to be clear enough to me... what's the purpose of key argument ?

File: app/src/wrapper/awsSdkWrapper.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+const AWS = require('aws-sdk');
```
Comment (przemos) [inline app/src/wrapper/awsSdkWrapper.js:1]: why is that required ?

File: app/src/wrapper/envVariablesWrapper.js [hunk 5]
```diff
@@ -0,0 +1 @@
+module.exports = process.env;
```
Comment (przemos) [inline app/src/wrapper/envVariablesWrapper.js:1]: is that to facilitate mocking ?

File: app/src/wrapper/awsSdkWrapper.js (hunk 4 above)
Comment (Marcin-Duszynski) [reply app/src/wrapper/awsSdkWrapper.js:1]: Because aws-sdk is saved as dev dependencies, all dependencies to aws-sdk in production code are marked as a error. That why this file is added to the esLint exclusion list. AWS-SDK is installed in AWS lambdas by default.

File: app/src/wrapper/envVariablesWrapper.js (hunk 5 above)
Comment (Marcin-Duszynski) [reply app/src/wrapper/envVariablesWrapper.js:1]: yes, to make tests independent from env

File: app/src/config/smmtConfigurationLoader.js (hunk 3 above)
Comment (Marcin-Duszynski) [reply app/src/config/smmtConfigurationLoader.js]: key parameter is used by unit tests. it allows to verify if KMS will not be called if key was already fetched.

**Review by ryszardm-kainos — APPROVED** [review]

**Review by przemos — APPROVED** [review]
